modules/git/batch_reader.go

```diff
@@ -269,6 +269,6 @@ func ParseTreeLine(rd *bufio.Reader, modeBuf, fnameBuf, shaBuf []byte) (mode, fn
 		}
 		idx += read
 	}
-	sha = shaBuf
+	sha = to40ByteSHA(shaBuf)
```

Comment:
This will drastically slow down the repository main page.

The comment explains very carefully why this is not done:

```go
// ParseTreeLine reads an entry from a tree in a cat-file --batch stream
// This carefully avoids allocations - except where fnameBuf is too small.
// It is recommended therefore to pass in an fnameBuf large enough to avoid almost all allocations
//
// Each line is composed of:
// <mode-in-ascii-dropping-initial-zeros> SP <fname> NUL <20-byte SHA>
//
// We don't attempt to convert the 20-byte SHA to 40-byte SHA to save a lot of time
```

Don't do this.